cmd/podman/search.go

```diff
@@ -0,0 +1,298 @@
+package main
+
+import (
+	"context"
+	"reflect"
+	"strconv"
+	"strings"
+
+	"github.com/containers/image/docker"
+	"github.com/pkg/errors"
+	"github.com/projectatomic/libpod/cmd/podman/formats"
+	"github.com/projectatomic/libpod/libpod"
+	"github.com/projectatomic/libpod/libpod/common"
+	"github.com/urfave/cli"
+)
+
+const (
+	descriptionTruncLength = 44
+	maxQueries             = 25
+)
+
+var (
+	searchFlags = []cli.Flag{
+		cli.StringSliceFlag{
+			Name:  "filter, f",
+			Usage: "filter output based on conditions provided (default [])",
+		},
+		cli.StringFlag{
+			Name:  "format",
+			Usage: "change the output format to a Go template",
+		},
+		cli.IntFlag{
+			Name:  "limit",
+			Usage: "limit the number of results",
+		},
+		cli.BoolFlag{
+			Name:  "no-trunc",
+			Usage: "do not truncate the output",
+		},
+		cli.StringFlag{
```

Comment:
Can/should we allow them to specify more than one registry?

Reply:
I don't see why not. This is a feature I added as podman search searches multiple registries based on how many are set as default registries in the config file. This was just a way to give the user to specify which registry he/she wants to search without dealing with the lag of searching all default registries.
If we want to allow specifying more than one, it can be done. @rhatdan what do you think?

Reply:
The config file being /etc/containers/registries.conf

Reply:
Yes allow me to specify multiple registries, I can even specify ones that are not in registries.conf

Reply:
fixed